Algorithm.CSharp/Alphas/TripleLeveragedETFPairVolatilityDecayAlphaAlgorithm.cs

```diff
@@ -0,0 +1,98 @@
+    /// ETFs with equal weights each day.
+
+    /// <br><br>This alpha is part of the Benchmark Alpha Series created by QuantConnect which are open sourced so the community and client funds can see an example of an alpha. 
+    /// You can read the source code for this alpha on Github in <a href="https://github.com/QuantConnect/Lean/blob/master/Algorithm.CSharp/Alphas/TripleLeverageETFPairVolatilityDecayAlpha.cs">C#</a>
+    /// or <a href="https://github.com/QuantConnect/Lean/blob/master/Algorithm.Python/Alphas/TripleLeverageETFPairVolatilityDecayAlpha.py">Python</a>
+    /// </summary>
+    public class TripleLeveragedETFPairVolatilityDecayAlphaAlgorithm : QCAlgorithmFramework
+    {
+        public override void Initialize()
+        {
+            SetStartDate(2018, 1, 1);
+
+            SetCash(100000);
+
+            // Set zero transaction fees
+            SetSecurityInitializer(security => security.FeeModel = new ConstantFeeModel(0));
+
+            // 3X ETF pair tickers
+            var ultraLong = QuantConnect.Symbol.Create("UGLD", SecurityType.Equity, Market.USA);
+            var ultraShort = QuantConnect.Symbol.Create("DGLD", SecurityType.Equity, Market.USA);
+
+            // Manually curated universe
+            UniverseSettings.Resolution = Resolution.Daily;
+            SetUniverseSelection(new ManualUniverseSelectionModel(new[] { ultraLong, ultraShort }));
+
+            // Select the demonstration alpha model
+            SetAlpha(new RebalancingTripleLeveragedETFAlphaModel(ultraLong, ultraShort));
+
+            // Select our default model types
+            SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
+        }
+
+        /// <summary>
+        /// Rebalance a pair of 3x leveraged ETFs and predict that the value of both ETFs in each pair will decrease.
+        /// </summary>
+        private class RebalancingTripleLeveragedETFAlphaModel : AlphaModel
+        {
+            private const double _magnitude = 0.001;
+            private readonly Symbol _ultraLong;
+            private readonly Symbol _ultraShort;
+            private readonly TimeSpan _period;
+
+            public RebalancingTripleLeveragedETFAlphaModel(Symbol ultraLong, Symbol ultraShort)
+            {
+                // Giving an insight period 1 days.
+                _period = QuantConnect.Time.OneDay;
+
+                _ultraLong = ultraLong;
+                _ultraShort = ultraShort;
+
+                Name = "RebalancingTripleLeveragedETFAlphaModel";
+            }
+
+            public override IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slice data)
+            {
+                return Insight.Group(new[]
+                {
+                    Insight.Price(_ultraLong, _period, InsightDirection.Down, _magnitude),
+                    Insight.Price(_ultraShort, _period, InsightDirection.Down, _magnitude)
```

Comment:
Is there some logic missing here? It seems to me its always returning the same insights

Reply:
This is right because we want to rebalance the positions every day:
> this alpha emits short-biased insight to capitalize on volatility decay for each listed pair of TL-ETFs, by rebalancing the ETFs with equal weights each day.